admiral/pkg/clusters/envoyfilter.go

```diff
@@ -0,0 +1,153 @@
+package clusters
+
+import (
+	"fmt"
+	"github.com/gogo/protobuf/types"
+	v1 "github.com/istio-ecosystem/admiral/admiral/pkg/apis/admiral/v1"
+	"github.com/istio-ecosystem/admiral/admiral/pkg/controller/admiral"
+	"github.com/istio-ecosystem/admiral/admiral/pkg/controller/common"
+	log "github.com/sirupsen/logrus"
+	"istio.io/api/networking/v1alpha3"
+	networking "istio.io/client-go/pkg/apis/networking/v1alpha3"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"strings"
+)
+
+func createOrUpdateEnvoyFilter( rc *RemoteController, routingPolicy *v1.RoutingPolicy, eventType admiral.EventType, workloadIdentityKey string, admiralCache *AdmiralCache) (*networking.EnvoyFilter, error) {
+
+	workloadSelectors := make(map[string]string)
+	workloadSelectors[common.GetWorkloadIdentifier()] = workloadIdentityKey
+	workloadSelectors[common.GetEnvKey()] = common.GetRoutingPolicyEnv(routingPolicy)
+
+	envoyfilterSpec := constructEnvoyFilterStruct(routingPolicy, workloadSelectors)
+
+	selectorLabelsSha, err := common.GetSha1(workloadIdentityKey+common.GetRoutingPolicyEnv(routingPolicy))
+	if err != nil {
+		log.Error("Error ocurred while computing workload Labels sha1")
+		return nil, err
+	}
+	envoyFilterName := fmt.Sprintf("%s-dynamicrouting-%s-%s", strings.ToLower(routingPolicy.Spec.Plugin), selectorLabelsSha, "1.10")
+	envoyfilter := &networking.EnvoyFilter{
+		TypeMeta: metaV1.TypeMeta{
+			Kind:       "EnvoyFilter",
+			APIVersion: "networking.istio.io/v1alpha3",
+		},
+		ObjectMeta: metaV1.ObjectMeta{
+			Name:      envoyFilterName,
+			Namespace: "istio-system",
+		},
+		Spec: *envoyfilterSpec,
+	}
+
+	admiralCache.RoutingPolicyFilterCache.Put(workloadIdentityKey+common.GetRoutingPolicyEnv(routingPolicy), rc.ClusterID, envoyFilterName)
+	var filter *networking.EnvoyFilter
+	//get the envoyfilter if it exists. If it exists, update it. Otherwise create it.
+	if eventType == admiral.Add || eventType == admiral.Update {
+		filter, err = rc.RoutingPolicyController.IstioClient.NetworkingV1alpha3().EnvoyFilters("istio-system").Get(envoyFilterName, metaV1.GetOptions{})
+		if err != nil {
+			log.Infof("msg=%s filtername=%s clustername=%s", "creating the envoy filter", envoyFilterName, rc.ClusterID)
+			filter, err = rc.RoutingPolicyController.IstioClient.NetworkingV1alpha3().EnvoyFilters("istio-system").Create(envoyfilter)
+			if err != nil {
+				log.Infof("Error creating filter: %v", filter)
+			}
+		} else {
+			log.Infof("msg=%s filtername=%s clustername=%s", "updating existing envoy filter", envoyFilterName, rc.ClusterID)
+			envoyfilter.ResourceVersion = filter.ResourceVersion
+			filter, err = rc.RoutingPolicyController.IstioClient.NetworkingV1alpha3().EnvoyFilters("istio-system").Update(envoyfilter)
+		}
+	}
```

Comment:
```suggestion
	}else{
	log.Infof("Unsupported event type %s yet", eventType )
	}
```

Reply:
The event type will not be anything other than these 2 if the control reaches in the envoyfilters part.